src/screens/CreateAccount/CreateAccount.tsx

```diff
@@ -0,0 +1,166 @@
+import { useState } from "react";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+
+import { SignUp, Logo, Letter, Lock, User } from "../../assets";
+import { Input, RegistrationMessage } from "../../components";
+
+const rMsg = "Field is Required";
+const permittedDomains = ["hotmail.com", "gmail.com", "lightit.io"];
+
+export const userSchema = z
+  .object({
+    email: z
+      .string()
+      .min(1, rMsg)
+      .email({
+        message: "Must be a valid email",
+      })
+      .refine(
+        (email) => {
+          const domain = email.split("@")[1];
+          return permittedDomains.includes(domain);
+        },
+        { message: "Email domain is not permitted" }
+      ),
+    username: z
+      .string()
+      .min(1, rMsg)
+      .refine((username) => !/\s/.test(username), {
+        message: "Username cannot contain spaces",
+      })
+      .refine((username) => !/\d/.test(username), {
+        message: "Username cannot contain numbers",
+      })
+      .refine((username) => /^[a-z]+$/i.test(username), {
+        message: "Username cannot contain special characters",
+      })
+      .refine((username) => /^[a-z]+$/.test(username), {
+        message: "Username must contain only lowercase letters",
+      }),
+    password: z
+      .string()
+      .min(8, { message: "Password must be at least 8 characters" })
+      .refine((password) => !/^[a-zA-Z0-9]+$/i.test(password), {
+        message: "Password must contain at least one special character",
+      })
+      .refine((password) => /^(?=.*[a-z])(?=.*[A-Z])/.test(password), {
+        message:
+          "Password must contain at least one lowercase and one uppercase letter",
+      }),
+    confirm_password: z.string().min(1, rMsg),
+  })
+  .refine((data) => data.password === data.confirm_password, {
+    path: ["confirm_password"],
+    message: "Password don't match",
+  });
+
+export type UserFormValues = z.infer<typeof userSchema>;
+
+export const CreateAccount = () => {
```

Comment:
I would name it `CreateAccountScreen`, since the current name doesn't tell you whether it is a button, a screen, a form, etc. But up to you.